src/main/java/gregtech/client/model/BorderlessLampBakedModel.java

```diff
@@ -0,0 +1,40 @@
+package gregtech.client.model;
+
+import gregtech.client.utils.RenderUtil;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.util.EnumFacing;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BorderlessLampBakedModel extends LampBakedModel {
+
+    // for each 6 side plus "no face" quads
+    @SuppressWarnings("unchecked")
+    private final List<BakedQuad>[] quads = new List[7];
+
+    public BorderlessLampBakedModel(ModelResourceLocation modelLocation) {
+        super(modelLocation);
+    }
+
+    public BorderlessLampBakedModel(IBakedModel model) {
+        super(model);
+    }
+
+    @Override
+    public List<BakedQuad> getQuads(@Nullable IBlockState state, @Nullable EnumFacing side, long rand) {
```

Comment:
Why doesn't this have to go through the same filtering system as the non borderless model?

Reply:
Bordered lamps have two layers to draw (SOLID for border/base, BLOOM or CUTOUT for emissive layer). Borderless blocks only have one layer to draw (BLOOM/SOLID). So it's more optimal to treat borderless lamp separately.